From f89821962898966ed78586832e1a39d89aecd259 Mon Sep 17 00:00:00 2001
From: daviiid99 <daviiideveloper@gmail.com>
Date: Thu, 25 Aug 2022 12:41:02 +0200
Subject: [PATCH] whole comment gpuservice file

---
 services/gpuservice/gpuservice.rc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/services/gpuservice/gpuservice.rc b/services/gpuservice/gpuservice.rc
index 0da8bd3812..b8df289255 100644
--- a/services/gpuservice/gpuservice.rc
+++ b/services/gpuservice/gpuservice.rc
@@ -1,4 +1,4 @@
-service gpu /system/bin/gpuservice
-    class core
-    user gpu_service
-    group graphics readtracefs
+#service gpu /system/bin/gpuservice
+#    class core
+#    user gpu_service
+#    group graphics readtracefs
-- 
2.34.1

