From 86e86f5a4d1d12327493b616bfdc2c682a4ca077 Mon Sep 17 00:00:00 2001
From: daviiid99 <daviiideveloper@gmail.com>
Date: Mon, 19 Sep 2022 17:12:21 +0200
Subject: [PATCH] Update camera_in_mediaserver_defaults

---
 build/soong/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/soong/Android.bp b/build/soong/Android.bp
index 34af118c..574968a4 100644
--- a/build/soong/Android.bp
+++ b/build/soong/Android.bp
@@ -295,6 +295,7 @@ camera_in_mediaserver {
                 "android.hardware.camera.common@1.0",
                 "android.hardware.camera.device@3.4",
                 "android.hardware.camera.provider@2.6",
+                "android.hardware.camera.provider@2.7",
                 "libcameraservice",
                 "libgui",
             ],
-- 
2.34.1

